src/main/java/magicbees/tileentity/RingHousing.java

```diff
@@ -0,0 +1,232 @@
+package magicbees.tileentity;
+
+import java.io.DataInputStream;
+import java.io.DataOutputStream;
+import java.io.IOException;
+
+import magicbees.bees.BeeManager;
+import magicbees.itemInventories.InventoryBeeRing;
+
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.ChunkCoordinates;
+import net.minecraft.util.Vec3;
+import net.minecraft.world.World;
+import net.minecraft.world.biome.BiomeGenBase;
+
+import com.google.common.collect.ImmutableSet;
+import com.mojang.authlib.GameProfile;
+
+import forestry.api.apiculture.*;
+import forestry.api.core.EnumHumidity;
+import forestry.api.core.EnumTemperature;
+import forestry.api.core.IErrorLogic;
+import forestry.api.core.IErrorState;
+
+// Class that simulates a BeeHouse using both player data and info from InventoryBeeRing
+public class RingHousing implements IBeeHousing {
+
+    EntityPlayer player;
+    BiomeGenBase biome;
+    IBeeHousingInventory inventory;
+    IBeekeepingLogic beekeepingLogic;
```

Comment:
Express access modifiers explicitly.

Reply:
Fixed. Fields are now private and final.